presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java

```diff
@@ -444,13 +450,15 @@ private void writeChunkToOutputStream(byte[] chunk, int offset, int length)
 
         int compressedSize = compressor.compress(chunk, offset, length, compressionBuffer, 0, compressionBuffer.length);
         if (compressedSize < length) {
+            assert PAGE_HEADER_SIZE + compressedSize <= maxBufferSize;
```

Comment:
The assert should be always on the length <= maxBufferSize. Checking on compressed size is incorrect as compressed size will always be less than length. maxBufferSize already accounts for PAGE_HEADER_SIZE and using that in addition to length will result in incorrect assertions.

Reply:
Good catch. This is from the previous approach that I initially drafted.